src/main.rs

```diff
@@ -134,7 +134,7 @@ fn real_main(buf: Rc<str>, bin_opt: BinOpt, output: &Path) -> Result<(), (Error,
         let stdout = io::stdout();
         let mut stdout_buf = BufWriter::new(stdout.lock());
         for token in rcc_try!(tokens, files) {
-            write!(stdout_buf, "{} ", token.data).expect("failed to write to stdout");
+            write!(stdout_buf, "{}", token.data).expect("failed to write to stdout");
         }
         writeln!(stdout_buf).expect("failed to write to stdout");
```

Comment:
Presumably the original C file ended with a newline so this would maintain the ending newline. What does this change do?

Reply:
Before the output for `-E` looked like this:
```
int main() {
	return 1;
}

```
Now it will look like this:
```
int main() {
	return 1;
}
```

Reply:
In other words, originally you would print both the original ending newline and add a newline of your own.